concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,479 @@
+    WHEN electivesurgery1 IS NULL THEN 6
+    WHEN adm_elective1 = 0 THEN 6
+    ELSE 0
+    END AS electivesurgery_OASIS
+  FROM elective_surgery
+)
+
+-- Heart rate 
+, heartrate_OASIS AS (
+SELECT patientunitstayid AS pid_HR
+  , CASE
+    WHEN MIN(heartrate) < 33 THEN 4
+    WHEN MAX(heartrate) BETWEEN 33 AND 88 THEN 0
+    WHEN MAX(heartrate) BETWEEN 89 AND 106 THEN 1
+    WHEN MAX(heartrate) BETWEEN 107 AND 125 THEN 3
+    WHEN MAX(heartrate) > 125 THEN 6
+    ELSE NULL
+    END AS heartrate_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND heartrate IS NOT NULL
+  GROUP BY pid_HR
+)
+
+-- Mean arterial pressure
+, MAP_OASIS AS (
+  SELECT patientunitstayid AS pid_MAP
+  , CASE
+    WHEN MIN(ibp_mean) < 20.65 THEN 4
+    WHEN MIN(ibp_mean) BETWEEN 20.65 AND 50.99 THEN 3
+    WHEN MIN(ibp_mean) BETWEEN 51 AND 61.32 THEN 2
+    WHEN MIN(ibp_mean) BETWEEN 61.33 AND 143.44 THEN 0
+    WHEN MAX(ibp_mean) >143.44 THEN 3
+    
+    WHEN MIN(nibp_mean) < 20.65 THEN 4
+    WHEN MIN(nibp_mean) BETWEEN 20.65 AND 50.99 THEN 3
+    WHEN MIN(nibp_mean) BETWEEN 51 AND 61.32 THEN 2
+    WHEN MIN(nibp_mean) BETWEEN 61.33 AND 143.44 THEN 0
+    WHEN MAX(nibp_mean) >143.44 THEN 3
+    ELSE NULL
+    END AS MAP_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+  GROUP BY pid_MAP
+)
+
+-- Respiratory rate
+, respiratoryrate_OASIS AS (
+SELECT patientunitstayid AS pid_RR
+  , CASE
+    WHEN MIN(respiratoryrate) < 6 THEN 10
+    WHEN MIN(respiratoryrate) BETWEEN 6 AND 12 THEN 1
+    WHEN MIN(respiratoryrate) BETWEEN 13 AND 22 THEN 0
+    WHEN MAX(respiratoryrate) BETWEEN 23 AND 30 THEN 1
+    WHEN MAX(respiratoryrate) BETWEEN 31 AND 44 THEN 6
+    WHEN MAX(respiratoryrate) > 44 THEN 9
+    ELSE NULL
+    END AS respiratoryrate_OASIS
+
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND respiratoryrate IS NOT NULL
+  GROUP BY pid_RR
+)
+
+-- Temperature 
+, temperature_OASIS AS (
+  SELECT patientunitstayid AS pid_temp
+  , CASE
+    WHEN MIN(temperature) < 33.22 THEN 3
+    WHEN MIN(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MAX(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MIN(temperature) BETWEEN 35.94 AND 36.39 THEN 2
+    WHEN MAX(temperature) BETWEEN 36.40 AND 36.88 THEN 0
+    WHEN MAX(temperature) BETWEEN 36.89 AND 39.88 THEN 2
+    WHEN MAX(temperature) >39.88 THEN 6
+    ELSE NULL
+    END AS temperature_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND temperature IS NOT NULL
+  GROUP BY pid_temp
+)
+
+-- Urine output
+, urineoutput_OASIS AS (
+  SELECT patientunitstayid AS pid_urine
+  , CASE
+    WHEN SUM(urineoutput) <671 THEN 10
```

Comment:
Urine output is notoriously poorly documented. I worry that a rule like this will find a single instance of documented UO for a day and score the patient very poorly. It's worth seeing what the distribution of scores look like for UO. For MIMIC, I had to do a lot of rules which roughly equated to "check that they are documenting UO regularly, and if they are, sum it up for the day". Sometimes this involved doing a sum over an ~8 hour period, and assuming that rate was consistent for the day.

I think if you compare this summation with the value stored in *apacheApsVar* (`urine`), you might get a better idea for any inconsistencies in this tabulation. In fact using that `urine` column instead might be a better bet.

Reply:
I have the impression that urine from apacheapsvar is even less reliable. I have sent you an email with some summary statistics. What do you think?

Reply:
I'll take a look!